x/operator/keeper/consensus_keys.go

```diff
@@ -563,3 +570,122 @@ func (k *Keeper) GetAllOperatorConsKeyRecords(ctx sdk.Context) ([]types.Operator
+	val.ConsAddress = consAddr.String()
+	val.Commission = ops.Commission
+
+	assets, err := k.avsKeeper.GetAVSSupportedAssets(ctx, avsAddrStr)
+	if err != nil {
+		return types.Validator{}, false
+	}
+	if assets == nil {
+		return types.Validator{}, false
+	}
+	// get the prices and decimals of assets
+	decimals, err := k.assetsKeeper.GetAssetsDecimal(ctx, assets)
+	if err != nil {
+		return types.Validator{}, false
+	}
+	prices, err := k.oracleKeeper.GetMultipleAssetsPrices(ctx, assets)
+	// TODO: for now, we ignore the error when the price round is not found and set the price to 1 to avoid panic
+	if err != nil {
+		// TODO: when assetID is not registered in oracle module, this error will finally lead to panic
+		if !errors.Is(err, oracletype.ErrGetPriceRoundNotFound) {
+			ctx.Logger().Error("fail to get price from oracle, since current assetID is not bonded with oracle token", "details:", err)
+			return types.Validator{}, false
+		}
+		// TODO: for now, we ignore the error when the price round is not found and set the price to 1 to avoid panic
+	}
+
+	ret := types.OperatorStakingInfo{
+		Staking:                 sdkmath.LegacyNewDec(0),
+		SelfStaking:             sdkmath.LegacyNewDec(0),
+		StakingAndWaitUnbonding: sdkmath.LegacyNewDec(0),
+	}
+	delegatorTokens := make([]types.DelegatorInfo, 0)
+
+	opFunc := func(assetID string, state *assetstypes.OperatorAssetInfo) error {
+		price, ok := prices[assetID]
+		if !ok {
+			return errorsmod.Wrap(types.ErrKeyNotExistInMap, "CalculateUSDValueForOperator map: prices, key: assetID")
+		}
+		decimal, ok := decimals[assetID]
+		if !ok {
+			return errorsmod.Wrap(types.ErrKeyNotExistInMap, "CalculateUSDValueForOperator map: decimals, key: assetID")
+		}
+		ret.Staking = ret.Staking.Add(CalculateUSDValue(state.TotalAmount, price.Value, decimal, price.Decimal))
+		// calculate the token amount from the share for the operator
+		selfAmount, err := delegationkeeper.TokensFromShares(state.OperatorShare, state.TotalShare, state.TotalAmount)
+		if err != nil {
+			return err
+		}
+		ret.SelfStaking = ret.SelfStaking.Add(CalculateUSDValue(selfAmount, price.Value, decimal, price.Decimal))
+		assetInfo, err := k.assetsKeeper.GetStakingAssetInfo(ctx, assetID)
+		if err != nil {
+			return err
+		}
+
+		info := types.DelegatorInfo{
+			AssetID:       assetID,
+			Symbol:        assetInfo.AssetBasicInfo.Symbol,
+			Name:          assetInfo.AssetBasicInfo.Name,
+			Amount:        state.TotalAmount,
+			TotalUSDValue: ret.Staking,
+		}
+		delegatorTokens = append(delegatorTokens, info)
+
+		return nil
+	}
+
+	if err := k.assetsKeeper.IterateAssetsForOperator(ctx, false, operatorAddr.String(), assets, opFunc); err != nil {
+		ctx.Logger().Error("IterateAssetsForOperator error", "err", err)
+		return types.Validator{}, false
+	}
+	val.VotingPower = ret.Staking
+	val.DelegatorShares = ret.Staking.Sub(ret.SelfStaking).TruncateInt()
+	val.DelegatorTokens = delegatorTokens
+
+	return val, true
+}
```

Comment:
The code in this function looks copied from other modules / files. Is it possible to reuse the code from them instead of duplicating it here?

Reply:
At present, it appears that it cannot be reused